File: pset6/vigenere.py
import sys
import cs50

if len(sys.argv)!=2:
    print("usage: vigenere.py k, where k is a string.")
    sys.exit(1)
for c in sys.argv[1]:
    if c.isalpha() == False:
        print("the key must be alphabetical characters only.")
        sys.exit(2)

key = sys.argv[1]

print("plaintext: ", end="")
text = cs50.get_string()

print("ciphertext: ", end="")

j=0
m=len(key)

for c in text:
    if key[j].isupper():
        if ord(c) >= 65 and ord(c) <= 90:
            print(chr(((ord(c)-65+ord(key[j])-65)%26)+65), end="")
        elif ord(c) >= 97 and ord(c) <= 122:
            print(chr(((ord(c)-97+ ord(key[j])-65)%26)+97), end="")
        else:
            print(c, end="")
    else:
        if ord(c) >= 65 and ord(c) <= 90:
            print(chr(((ord(c)-65+ord(key[j])-97)%26)+65), end="")
        elif ord(c) >= 97 and ord(c) <= 122:
            print(chr(((ord(c)-97+ ord(key[j])-97)%26)+97), end="")
        else: 
            print(c, end="")
        
    if c.isalpha():
        j = (j+1)% m
    
print("")

sys.exit(0)

File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if(argc != 2)
    {
        printf("usage: ./vigenere k, where k is a string of alphabetical characters.\n");
        return 1;
    }
    for(int i = 0, n= strlen(argv[1]);i<n;i++)
    {
        if(isalpha(argv[1][i])== false)
        {
            printf("the key must be a string of alphabetical characters only.\n");
            return 1;
        }
    }
    
    string key = argv[1];
    
    printf("plaintext: ");
    string text = get_string();
    
    int j = 0;
    int m = strlen(key);
    
    printf("ciphertext: ");
    
    for (int i = 0, n = strlen(text); i < n ; i++)
        {
            if (isupper (key[j]))
            {
                if (text[i] >= 65 && text[i] <= 90)
                printf("%c", ((text[i]-65+key[j]-65)%26)+65);
                else if (text[i] >= 97 && text[i] <= 122)
                printf("%c", ((text[i]-97+ key[j]-65)%26)+97);
                else printf("%c", text[i]);
            }
            else 
            {
                if (text[i] >= 65 && text[i] <= 90)
                printf("%c", ((text[i]-65+key[j]-97)%26)+65);
                else if (text[i] >= 97 && text[i] <= 122)
                printf("%c", ((text[i]-97+ key[j]-97)%26)+97);
                else printf("%c", text[i]);
            }
            if (isalpha (text[i]))
            j = (j+1)% m; 
        }
    printf("\n");
    
    
    
    return 0;
}